# EDA Dataset: Pizza Place Sales (Kaggle)
## Análisis de criterios de selección

Este notebook recoge el análisis de los criterios de selección del 
dataset Pizza Place Sales como dataset principal para el TFG de 
predicción de demanda en pizzerías con delivery.

In [3]:
import pandas as pd

# Carga del dataset
orders = pd.read_csv('../data/kaggle_pizza_place_sales/orders.csv')
order_details = pd.read_csv('../data/kaggle_pizza_place_sales/order_details.csv')

print("=== CRITERIO 1: VOLUMEN ===")
print(f"Registros en orders: {orders.shape[0]}")
print(f"Registros en order_details: {order_details.shape[0]}")


=== CRITERIO 1: VOLUMEN ===
Registros en orders: 21350
Registros en order_details: 48620


## Criterio 1: volumen
El dataset tiene un tamañao adecuado para entrenar el modelo.

In [5]:
print("\n=== CRITERIO 2: GRANULARIDAD TEMPORAL ===")
print(f"Columnas disponibles: {orders.columns.tolist()}")
print(f"¿Tiene fecha? {'date' in orders.columns}")
print(f"¿Tiene hora? {'time' in orders.columns}")


=== CRITERIO 2: GRANULARIDAD TEMPORAL ===
Columnas disponibles: ['order_id', 'date', 'time']
¿Tiene fecha? True
¿Tiene hora? True


## Criterio 2: Fecha y hora
Los datos almacenados disponen de información sobre la fecha y hora en la que se realizaron.

In [6]:
print("\n=== CRITERIO 3: RANGO TEMPORAL ===")
orders['date'] = pd.to_datetime(orders['date'])
print(f"Rango: {orders['date'].min()} → {orders['date'].max()}")
print(f"Días cubiertos: {(orders['date'].max() - orders['date'].min()).days}")


=== CRITERIO 3: RANGO TEMPORAL ===
Rango: 2015-01-01 00:00:00 → 2015-12-31 00:00:00
Días cubiertos: 364


## Criterio 3: Rango temporal
Los días cubiertos permiten analizar un año entero.

In [7]:
print("\n=== CRITERIO 4: CALIDAD DE DATOS ===")
print(f"Valores nulos en orders: {orders.isnull().sum().sum()}")
print(f"Duplicados en orders: {orders.duplicated().sum()}")


=== CRITERIO 4: CALIDAD DE DATOS ===
Valores nulos en orders: 0
Duplicados en orders: 0


## Criterio 4: Calidad de los datos
Los registros del dataset no presentan valores nulos ni registros duplicados.

In [8]:
print("=== CRITERIO 5: ESPECIFICIDAD DE RESTAURACIÓN ===")
pizza_types = pd.read_csv('../data/kaggle_pizza_place_sales/pizza_types.csv', 
                          encoding='latin-1')
print(f"Tipos de pizza: {pizza_types.shape[0]}")
print(f"Categorías: {pizza_types['category'].unique()}")
print(f"\nEste dataset corresponde a una única pizzería real.")

=== CRITERIO 5: ESPECIFICIDAD DE RESTAURACIÓN ===
Tipos de pizza: 32
Categorías: <StringArray>
['Chicken', 'Classic', 'Supreme', 'Veggie']
Length: 4, dtype: str

Este dataset corresponde a una única pizzería real.


## Criterio 5: específico de restauración
En el dataset solo hay registros de pedidos de pizza.

## Verificación de criterios de selección

- ✅ **Volumen:** 21.350 pedidos registrados, 
  suficiente para entrenar un modelo de series temporales.
- ✅ **Fecha y hora:** fecha y hora exacta de cada pedido. 
  exacta de cada pedido.
- ✅ **Rango temporal:** cubre 364 días, prácticamente un año 
  completo (2015).
- ✅ **Calidad datos:** sin valores nulos ni duplicados.
- ✅ **Especificidad:** sí, solo contiene pedidos de pizza. 
  Los datos son reales de una única pizzería con   
  32 tipos de pizza en 4 categorías.


## Conclusión: dataset seleccionado

El análisis confirma que este dataset cumple todos los criterios 
de selección establecidos.

Aunque el dataset no incluye variables contextuales como clima o festivos, esto no supone ningún
problema, ya que la adición de esta información con APIs externas forma parte del desarrollo del proyecto y constituye una de las aportaciones principales del trabajo.

**Decisión: dataset seleccionado como base del proyecto.**